File: backend/app/schemas/ocr.py
"""ocr.py — OCR Pipeline Schemas."""

from datetime import datetime
from typing import Any
from pydantic import BaseModel, ConfigDict, Field

from app.models.quality import OcrJobStatus


class OcrJobCreate(BaseModel):
    document_id: int
    project_id: int


class OcrJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    job_id: str
    project_id: int
    document_id: int | None
    uploaded_by: int | None
    status: OcrJobStatus
    doc_type: str | None
    page_count: int | None
    is_digital: bool | None
    result_json: dict[str, Any] | None
    overall_confidence: float | None
    error_message: str | None
    processing_ms: int | None
    created_at: datetime
    completed_at: datetime | None


class OcrFieldCorrectionCreate(BaseModel):
    doc_type: str
    original_fields: dict[str, Any]
    final_fields: dict[str, Any]

File: frontend/react-app/src/types/ocr.ts
export type OcrJobStatus = 'QUEUED' | 'PROCESSING' | 'COMPLETED' | 'FAILED';

export interface OcrJobCreate {
  document_id: number;
  project_id: number;
}

export interface OcrJobResponse {
  job_id: string;
  project_id: number;
  document_id: number | null;
  uploaded_by: number | null;
  status: OcrJobStatus;
  doc_type: string | null;
  page_count: number | null;
  is_digital: boolean | null;
  result_json: {
    fields: Record<string, { value: any; confidence: number; method: string }>;
    domain_data: Record<string, any>;
    tables: any[][][];
    raw_text: string;
  } | null;
  overall_confidence: number | null;
  error_message: string | null;
  processing_ms: number | null;
  created_at: string;
  completed_at: string | null;
}

export interface OcrFieldCorrectionCreate {
  doc_type: string;
  original_fields: Record<string, any>;
  final_fields: Record<string, any>;
}
